src/java.base/share/classes/java/nio/channels/Channels.java

```diff
@@ -136,47 +136,66 @@ public static InputStream newInputStream(ReadableByteChannel ch) {
      */
     public static OutputStream newOutputStream(WritableByteChannel ch) {
         Objects.requireNonNull(ch, "ch");
+        return new ChannelOutputStream(ch);
+    }
 
-        return new OutputStream() {
+    /**
+     * @since 18
+     */
+    public static class ChannelOutputStream extends OutputStream {
```

Comment:
You mean as a source comment or just here in this discussion thread?

In fact it might be better to not add it to a package with is part of the API, but to move it to the `sun` package, which is not, right?

The justification is that I need to refer to this class from `ChannelInputStream::transferTo()` to be able to get hold of this previously anonymous class's inner member "ch", which in turn is key to the whole story of this PR: Using NIO transfer between the channels instead of moving bytes through the JVM's memory.

Reply:
Will move the class (and the needed utility methods) to the `sun` package to prevent addition to the API. Stay tuned. :-)

Reply:
`ChannelOutputStream` isn't part of the API anymore as I moved it to the `sun` package by commit https://github.com/openjdk/jdk/pull/4263/commits/b6dc6a7eadb5168531459aa032f169b344cadb47. Thankyou for pointing this out!